services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/configuration/Features.kt

```diff
@@ -0,0 +1,28 @@
+package fi.hel.haitaton.hanke.configuration
+
+import fi.hel.haitaton.hanke.EndpointDisabledException
+import org.springframework.boot.context.properties.ConfigurationProperties
+import org.springframework.boot.context.properties.ConstructorBinding
+
+@ConfigurationProperties(prefix = "haitaton")
+@ConstructorBinding
+data class Features(val features: Map<Feature, Boolean>) {
+
+    /** Disabled by default, if not in application.properties. */
+    private fun isEnabled(feature: Feature): Boolean = features.getOrDefault(feature, false)
+
+    /**
+     * Throws an exception if the feature is not enabled.
+     *
+     * Disabled by default, if not in application.properties.
+     */
+    fun check(feature: Feature) {
```

Comment:
Check may be a bit vague. Could this be renamed for example checkIfEnabled etc? On the other hand javadoc seems to explain the purpose pretty well.